src/memory/split.jl

```diff
@@ -422,12 +422,19 @@ function free(ptr)
 end
 
 function reclaim(sz::Int=typemax(Int))
-    freed = 0
+    if !isempty(freed)
+        blocks = Set(freed)
```

Comment:
Is this because the same block might be twice in the `freed` list? Or did you just want a lightweight copy?

Reply:
`freed` might be modified concurrently (from a finalizer), so I just want a copy re. https://github.com/JuliaGPU/CuArrays.jl/blob/0e7f8a9fe69a5042ed542fe2b0a72d9c38a583a8/src/memory/split.jl#L336-L342
I guess I decided back then to use a `Set` because `incremental_compact!` pops values, but that might have been a relic of past developments (I should probably go through the different uses of containers now that the code has stabilized a bit).